loki/transformations/temporaries/pool_allocator.py

```diff
@@ -205,6 +207,7 @@ def __init__(
         self.directive = directive
         self.check_bounds = check_bounds
         self.cray_ptr_loc_rhs = cray_ptr_loc_rhs
+        self.stack_size_var_type = stack_size_var_type or self.stack_int_type_kind
```

Comment:
To avoid the misnomer, we could store here directly a type instead of the kind:

```suggestion
        self.stack_size_var_type = stack_size_var_type or SymbolAttributes(BasicType.INTEGER, kind=self.stack_int_type_kind)
```